sources/zafira-web/src/main/java/com/qaprosoft/zafira/web/TestController.java

```diff
@@ -174,21 +174,15 @@ public void deleteTest(@ApiParam(value = "Test ID", required = true) @PathVariab
     }
 
     @ApiResponseStatuses
-    @ApiOperation(value = "Create or update test work item", nickname = "createOrUpdateTestWorkItem", httpMethod = "POST", response = WorkItem.class)
+    @ApiOperation(value = "Link test work item", nickname = "linkWorkItem", httpMethod = "POST", response = WorkItem.class)
     @ApiImplicitParams({ @ApiImplicitParam(name = "Authorization", paramType = "header") })
     @PostMapping("/{id}/workitem")
-    public WorkItem createOrUpdateTestWorkItem(
-            @ApiParam(value = "Test ID", required = true) @PathVariable("id") long id,
-            @RequestBody WorkItem workItem
-    ) {
+    public WorkItem linkWorkItem(@PathVariable("id") long id, @RequestBody WorkItem workItem) {
         if (getPrincipalId() > 0) {
```

Comment:
I've just noticed that we use db entity (instead of DTO) to pass data, then update it in controller, then in service, and return in response. Seems like it should be refactored (and the other endpoints, returning WorkItem.class as well). @nsidorevich , what do you think?

Reply:
@tsvrko it`s true, but it is another one scope of work. We have a lot of dirty code in controllers like domain object instead of dto